spec/unit/pops/loaders/loader_paths_spec.rb

```diff
@@ -38,6 +38,7 @@ def foo; end;
     expect(effective_paths.size).to eq(1)
     expect(module_loader.path_index.size).to eq(1)
     path_index = module_loader.path_index
-    expect(path_index).to include(File.join(module_dir, 'lib', 'puppet', 'functions', 'foo4x.rb'))
+    expected = File.join(module_dir, 'lib', 'puppet', 'functions', 'foo4x.rb')
+    expect(path_index).to include(Dir.glob(expected).first)
```

Comment:
Yeah, I'm still debating on whether this makes complete sense as implemented.

The way I've implemented things now, the behavior varies based on platform / Ruby version - which I'm honestly not an enormous fan of.  I'd rather do something to make the behavior consistent across at least the platform.